---
title: "Microbial Response to Intermittent Permafrost Thaw -- Correlation"
author: 'Authors: [Karl J. Romanowicz](https://lsa.umich.edu/eeb/people/graduate-students/kjromano.html) and George W. Kling'
output:
  html_notebook:
    theme: spacelab
    toc: yes
    toc_depth: 5
    toc_float:
      collapsed: yes
      smooth_scroll: yes
  html_document:
    toc: yes
    toc_depth: '5'
    df_print: paged
  pdf_document:
    toc: yes
    toc_depth: '5'
---

__________________________________________________

**R Notebook:** <font color="green">Provides reproducible analysis for **Thaw Duration Correlation** data in the following manuscript:</font>

**Citation:** Romanowicz, KJ and Kling, GW. (*In Review*) Annual thaw duration is a strong predictor of the soil microbiome and its response to permafrost thaw in arctic tundra.

**GitHub Repository:** [https://github.com/kromanowicz/2022-Annual-Thaw-Microbes](https://github.com/kromanowicz/2022-Annual-Thaw-Microbes)

**NCBI BioProject:** [https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA794857](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA794857)

# Experiment

This R Notebook provides complete reproducibility of the data analysis presented in ***"Annual thaw duration is a strong predictor of the soil microbiome and its response to permafrost thaw in arctic tundra"*** by Romanowicz and Kling. This pipeline processes 16S rRNA gene sequences that were generated using the Illumina MiSeq platform using paired-end sequencing read amplicons.

```{r setup, include=FALSE}
# Set global options for notebook
knitr::opts_knit$set(root.dir = normalizePath("~/Desktop/TundraPro18"))
knitr::opts_chunk$set(echo = TRUE)
```

```{r message=FALSE, results='hide', warning=FALSE}
# Make a vector of required packages
required.packages <- c("corrr","data.table","devtools","dplyr","forcats","ggalluvial","ggdendro","ggplot2","ggpubr","grid","gridExtra","knitr","magrittr","microeco","patchwork","pheatmap","pvclust","qiime2R","RColorBrewer","tidyr","UpSetR","vegan")

# Load required packages
lapply(required.packages, library, character.only = TRUE)
```

```{r include=FALSE}
# set.seed is used to fix the random number generation to make the results repeatable
set.seed(123)
```

# Correlations

Correlation analysis uses non-parametric Spearman Rank Correlations to overcome the lack of normally distributed data. Examples of normality (or lack thereof) within the data are provided in the **"Normality Check"** section of this workbook.

## Toolik MAT

### Thaw Days: Taxa
TTT Thaw Probability Correlations -- Soil Chemistry and Taxonomy
```{r}
ttt.data.corr<-read.csv("QIIME/R_Data/ttt.data.corr.csv")

# Convert first column to row names
rownames(ttt.data.corr)<-ttt.data.corr[,1]
ttt.data.corr<-ttt.data.corr[,-1]
```

"Thaw Days" as well as many taxa have non-normal distributions. Use Spearman's correlation as a non-parametric statistical analysis to overcome lack of normality.
```{r}
# Spearman Correlation Tests between Thaw Days and Taxonomy
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Acidobacteria, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Actinobacteria, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Bacteroidetes, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Caldiserica, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Chlorobi, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Chloroflexi, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Firmicutes, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Nitrospirae, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Planctomycetes, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Alphaproteobacteria, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Betaproteobacteria, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Deltaproteobacteria, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Gammaproteobacteria, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Verrucomicrobia, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Bacteria.Other, method="spearman")
cor.test(ttt.data.corr$Thaw_Days, ttt.data.corr$Archaea, method="spearman")
```

### Soil Chem: Taxa

Use Spearman's correlation for soil chemistry variables and taxa.
```{r}
# Correlation Tests between pH and Taxonomy
cor.test(ttt.data.corr$pH, ttt.data.corr$Acidobacteria, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Actinobacteria, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Bacteroidetes, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Caldiserica, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Chlorobi, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Chloroflexi, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Firmicutes, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Nitrospirae, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Planctomycetes, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Alphaproteobacteria, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Betaproteobacteria, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Deltaproteobacteria, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Gammaproteobacteria, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Verrucomicrobia, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Bacteria.Other, method="spearman")
cor.test(ttt.data.corr$pH, ttt.data.corr$Archaea, method="spearman")

# Correlation Tests between EC and Taxonomy
cor.test(ttt.data.corr$EC, ttt.data.corr$Acidobacteria, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Actinobacteria, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Bacteroidetes, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Caldiserica, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Chlorobi, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Chloroflexi, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Firmicutes, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Nitrospirae, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Planctomycetes, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Alphaproteobacteria, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Betaproteobacteria, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Deltaproteobacteria, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Gammaproteobacteria, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Verrucomicrobia, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Bacteria.Other, method="spearman")
cor.test(ttt.data.corr$EC, ttt.data.corr$Archaea, method="spearman")

# Correlation Tests between GWC and Taxonomy
cor.test(ttt.data.corr$GWC, ttt.data.corr$Acidobacteria, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Actinobacteria, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Bacteroidetes, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Caldiserica, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Chlorobi, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Chloroflexi, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Firmicutes, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Nitrospirae, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Planctomycetes, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Alphaproteobacteria, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Betaproteobacteria, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Deltaproteobacteria, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Gammaproteobacteria, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Verrucomicrobia, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Bacteria.Other, method="spearman")
cor.test(ttt.data.corr$GWC, ttt.data.corr$Archaea, method="spearman")

# Correlation Tests between TOC and Taxonomy
cor.test(ttt.data.corr$OC, ttt.data.corr$Acidobacteria, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Actinobacteria, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Bacteroidetes, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Caldiserica, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Chlorobi, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Chloroflexi, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Firmicutes, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Nitrospirae, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Planctomycetes, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Alphaproteobacteria, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Betaproteobacteria, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Deltaproteobacteria, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Gammaproteobacteria, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Verrucomicrobia, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Bacteria.Other, method="spearman")
cor.test(ttt.data.corr$OC, ttt.data.corr$Archaea, method="spearman")
```

### TTT Plotting

Before plotting, need to remove columns from full dataset
```{r}
# Remove thaw probability columns and soil chemistry for plotting
ttt.data.corr <- subset(ttt.data.corr, select = -c(Thaw_Prob_July_All, Thaw_Prob_Aug_All, Thaw_Prob_Aug_10, Thaw_Prob_Aug_5, pH, EC, GWC, OC))
```

```{r}
# Correlation plot for annual thaw duration (thaw days) using Spearman's rho
ttt.thaw.day.corr.plot <- ttt.data.corr %>% correlate(method = "spearman") %>% focus(Thaw_Days) %>% mutate(term = factor(term, levels = term[order(Thaw_Days)])) %>% ggplot(aes(x = term, y = Thaw_Days)) + geom_bar(stat = "identity") + theme_classic() + theme(axis.text.x = element_text(angle = 45, hjust=1), axis.title.x = element_blank(), text = element_text(size=16)) + ylab(expression(atop("Annual Thaw Duration", paste("Spearman Correlation")))) + ylim(-1,1)

ttt.thaw.day.corr.plot

# Save as .eps file (width = 600; height = 450; "ttt.thaw.day.corr.eps")
```

## Imnavait MAT

### Thaw Days: Taxa

ITT Thaw Probability Correlations -- Soil Chemistry and Taxonomy
```{r}
itt.data.corr<-read.csv("QIIME/R_Data/itt.data.corr.csv")

# Convert first column to row names
rownames(itt.data.corr)<-itt.data.corr[,1]
itt.data.corr<-itt.data.corr[,-1]
```

"Thaw Days" as well as many taxa have non-normal distributions. Use Spearman's correlation as a non-parametric statistical analysis to overcome lack of normality.
```{r}
# Spearman Correlation Tests between Thaw Days and Taxonomy
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Acidobacteria, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Actinobacteria, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Bacteroidetes, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Caldiserica, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Chlorobi, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Chloroflexi, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Firmicutes, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Nitrospirae, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Planctomycetes, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Alphaproteobacteria, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Betaproteobacteria, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Deltaproteobacteria, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Gammaproteobacteria, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Verrucomicrobia, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Bacteria.Other, method="spearman")
cor.test(itt.data.corr$Thaw_Days, itt.data.corr$Archaea, method="spearman")
```

### Soil Chem: Taxa

Use Spearman's correlation for soil chemistry variables and taxa.
```{r}
# Correlation Tests between pH and Taxonomy
cor.test(itt.data.corr$pH, itt.data.corr$Acidobacteria, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Actinobacteria, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Bacteroidetes, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Caldiserica, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Chlorobi, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Chloroflexi, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Firmicutes, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Nitrospirae, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Planctomycetes, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Alphaproteobacteria, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Betaproteobacteria, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Deltaproteobacteria, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Gammaproteobacteria, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Verrucomicrobia, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Bacteria.Other, method="spearman")
cor.test(itt.data.corr$pH, itt.data.corr$Archaea, method="spearman")

# Correlation Tests between EC and Taxonomy
cor.test(itt.data.corr$EC, itt.data.corr$Acidobacteria, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Actinobacteria, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Bacteroidetes, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Caldiserica, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Chlorobi, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Chloroflexi, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Firmicutes, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Nitrospirae, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Planctomycetes, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Alphaproteobacteria, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Betaproteobacteria, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Deltaproteobacteria, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Gammaproteobacteria, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Verrucomicrobia, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Bacteria.Other, method="spearman")
cor.test(itt.data.corr$EC, itt.data.corr$Archaea, method="spearman")

# Correlation Tests between GWC and Taxonomy
cor.test(itt.data.corr$GWC, itt.data.corr$Acidobacteria, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Actinobacteria, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Bacteroidetes, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Caldiserica, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Chlorobi, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Chloroflexi, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Firmicutes, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Nitrospirae, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Planctomycetes, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Alphaproteobacteria, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Betaproteobacteria, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Deltaproteobacteria, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Gammaproteobacteria, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Verrucomicrobia, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Bacteria.Other, method="spearman")
cor.test(itt.data.corr$GWC, itt.data.corr$Archaea, method="spearman")

# Correlation Tests between TOC and Taxonomy
cor.test(itt.data.corr$OC, itt.data.corr$Acidobacteria, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Actinobacteria, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Bacteroidetes, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Caldiserica, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Chlorobi, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Chloroflexi, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Firmicutes, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Nitrospirae, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Planctomycetes, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Alphaproteobacteria, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Betaproteobacteria, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Deltaproteobacteria, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Gammaproteobacteria, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Verrucomicrobia, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Bacteria.Other, method="spearman")
cor.test(itt.data.corr$OC, itt.data.corr$Archaea, method="spearman")
```

### ITT Plotting

Before plotting, need to remove columns from full dataset
```{r}
# Remove thaw probability columns and soil chemistry for plotting
itt.data.corr <- subset(itt.data.corr, select = -c(Thaw_Prob_July_All, Thaw_Prob_Aug_All, Thaw_Prob_Aug_10, Thaw_Prob_Aug_5, pH, EC, GWC, OC))
```

```{r}
# Correlation plot for annual thaw duration (thaw days) using Spearman's rho
itt.thaw.day.corr.plot <- itt.data.corr %>% correlate(method = "spearman") %>% focus(Thaw_Days) %>% mutate(term = factor(term, levels = term[order(Thaw_Days)])) %>% ggplot(aes(x = term, y = Thaw_Days)) + geom_bar(stat = "identity") + theme_classic() + theme(axis.text.x = element_text(angle = 45, hjust=1), axis.title.x = element_blank(), text = element_text(size=16)) + ylab(expression(atop("Annual Thaw Duration", paste("Spearman Correlation")))) + ylim(-1,1)

itt.thaw.day.corr.plot

# Save as .eps file (width = 600; height = 450; "itt.thaw.day.corr.eps")
```

## Imnavait WS

### Thaw Days: Taxa

IWS Thaw Probability Correlations -- Soil Chemistry and Taxonomy
```{r}
iws.data.corr<-read.csv("QIIME/R_Data/iws.data.corr.csv")

# Convert first column to row names
rownames(iws.data.corr)<-iws.data.corr[,1]
iws.data.corr<-iws.data.corr[,-1]
```

"Thaw Days" as well as many taxa have non-normal distributions. Use Spearman's correlation as a non-parametric statistical analysis to overcome lack of normality.
```{r}
# Spearman Correlation Tests between Thaw Days and Taxonomy
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Acidobacteria, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Actinobacteria, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Bacteroidetes, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Caldiserica, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Chlorobi, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Chloroflexi, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Firmicutes, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Nitrospirae, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Planctomycetes, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Alphaproteobacteria, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Betaproteobacteria, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Deltaproteobacteria, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Gammaproteobacteria, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Verrucomicrobia, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Bacteria.Other, method="spearman")
cor.test(iws.data.corr$Thaw_Days, iws.data.corr$Archaea, method="spearman")
```

### Soil Chem: Taxa

Use Spearman's correlation for soil chemistry variables and taxa.
```{r}
# Correlation Tests between pH and Taxonomy
cor.test(iws.data.corr$pH, iws.data.corr$Acidobacteria, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Actinobacteria, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Bacteroidetes, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Caldiserica, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Chlorobi, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Chloroflexi, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Firmicutes, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Nitrospirae, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Planctomycetes, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Alphaproteobacteria, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Betaproteobacteria, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Deltaproteobacteria, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Gammaproteobacteria, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Verrucomicrobia, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Bacteria.Other, method="spearman")
cor.test(iws.data.corr$pH, iws.data.corr$Archaea, method="spearman")

# Correlation Tests between EC and Taxonomy
cor.test(iws.data.corr$EC, iws.data.corr$Acidobacteria, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Actinobacteria, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Bacteroidetes, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Caldiserica, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Chlorobi, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Chloroflexi, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Firmicutes, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Nitrospirae, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Planctomycetes, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Alphaproteobacteria, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Betaproteobacteria, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Deltaproteobacteria, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Gammaproteobacteria, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Verrucomicrobia, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Bacteria.Other, method="spearman")
cor.test(iws.data.corr$EC, iws.data.corr$Archaea, method="spearman")

# Correlation Tests between GWC and Taxonomy
cor.test(iws.data.corr$GWC, iws.data.corr$Acidobacteria, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Actinobacteria, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Bacteroidetes, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Caldiserica, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Chlorobi, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Chloroflexi, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Firmicutes, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Nitrospirae, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Planctomycetes, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Alphaproteobacteria, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Betaproteobacteria, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Deltaproteobacteria, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Gammaproteobacteria, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Verrucomicrobia, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Bacteria.Other, method="spearman")
cor.test(iws.data.corr$GWC, iws.data.corr$Archaea, method="spearman")

# Correlation Tests between TOC and Taxonomy
cor.test(iws.data.corr$OC, iws.data.corr$Acidobacteria, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Actinobacteria, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Bacteroidetes, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Caldiserica, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Chlorobi, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Chloroflexi, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Firmicutes, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Nitrospirae, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Planctomycetes, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Alphaproteobacteria, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Betaproteobacteria, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Deltaproteobacteria, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Gammaproteobacteria, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Verrucomicrobia, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Bacteria.Other, method="spearman")
cor.test(iws.data.corr$OC, iws.data.corr$Archaea, method="spearman")
```

### IWS Plotting

Before plotting, need to remove columns from full dataset
```{r}
# Remove thaw probability columns and soil chemistry for plotting
iws.data.corr <- subset(iws.data.corr, select = -c(Thaw_Prob_July_All, Thaw_Prob_Aug_All, Thaw_Prob_Aug_10, Thaw_Prob_Aug_5, pH, EC, GWC, OC))
```

```{r}
# Correlation plot for annual thaw duration (thaw days) using Spearman's rho
iws.thaw.day.corr.plot <- iws.data.corr %>% correlate(method = "spearman") %>% focus(Thaw_Days) %>% mutate(term = factor(term, levels = term[order(Thaw_Days)])) %>% ggplot(aes(x = term, y = Thaw_Days)) + geom_bar(stat = "identity") + theme_classic() + theme(axis.text.x = element_text(angle = 45, hjust=1), axis.title.x = element_blank(), text = element_text(size=16)) + ylab(expression(atop("Annual Thaw Duration", paste("Spearman Correlation")))) + ylim(-1,1)

iws.thaw.day.corr.plot

# Save as .eps file (width = 600; height = 450; "iws.thaw.day.corr.eps")
```

# Normality Check

## Thaw Days

TTT Thaw Probability Correlations -- Soil Chemistry and Taxonomy
```{r}
ttt.data.corr<-read.csv("QIIME/R_Data/ttt.data.corr.csv")

# Convert first column to row names
rownames(ttt.data.corr)<-ttt.data.corr[,1]
ttt.data.corr<-ttt.data.corr[,-1]
```

Check for normal distribution of data prior to running parametric correlations
```{r}
#Thaw Days
shapiro.test(ttt.data.corr$Thaw_Days)

# Distribution of Thaw_Days variable
ggdensity(ttt.data.corr, x = "Thaw_Days", fill = "lightgray", title = "Thaw Days") + scale_x_continuous(limits = c(-100, 200)) + stat_overlay_normal_density(color = "red", linetype = "dashed")
```

Transform data for Normality
```{r}
#Thaw_Days
ttt.data.corr$Thaw_Days <- sqrt(ttt.data.corr$Thaw_Days)
```

```{r}
#Thaw Days
shapiro.test(ttt.data.corr$Thaw_Days)

# Distribution of CONT variable
ggdensity(ttt.data.corr, x = "Thaw_Days", fill = "lightgray", title = "Thaw_Days") + scale_x_continuous(limits = c(-10, 20)) + stat_overlay_normal_density(color = "red", linetype = "dashed")
```

## Taxonomy

```{r}
#Taxonomy
shapiro.test(ttt.data.corr$Acidobacteria)
shapiro.test(ttt.data.corr$Actinobacteria)
shapiro.test(ttt.data.corr$Bacteroidetes)
shapiro.test(ttt.data.corr$Caldiserica)
shapiro.test(ttt.data.corr$Chlorobi)
shapiro.test(ttt.data.corr$Chloroflexi)
shapiro.test(ttt.data.corr$Firmicutes)
shapiro.test(ttt.data.corr$Nitrospirae)
shapiro.test(ttt.data.corr$Planctomycetes)
shapiro.test(ttt.data.corr$Alphaproteobacteria)
shapiro.test(ttt.data.corr$Betaproteobacteria)
shapiro.test(ttt.data.corr$Deltaproteobacteria)
shapiro.test(ttt.data.corr$Gammaproteobacteria)
shapiro.test(ttt.data.corr$Verrucomicrobia)
shapiro.test(ttt.data.corr$Bacteria.Other)
shapiro.test(ttt.data.corr$Archaea)

# Distribution of Taxonomy variables
ggdensity(ttt.data.corr, x = "Acidobacteria", fill = "lightgray", title = "Acidobacteria") + scale_x_continuous(limits = c(-50, 50)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Actinobacteria", fill = "lightgray", title = "Actinobacteria") + scale_x_continuous(limits = c(-25, 75)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Bacteroidetes", fill = "lightgray", title = "Bacteroidetes") + scale_x_continuous(limits = c(-50, 50)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Caldiserica", fill = "lightgray", title = "Caldiserica") + scale_x_continuous(limits = c(-25, 25)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Chlorobi", fill = "lightgray", title = "Chlorobi") + scale_x_continuous(limits = c(-5, 5)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Chloroflexi", fill = "lightgray", title = "Chloroflexi") + scale_x_continuous(limits = c(-25, 25)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Firmicutes", fill = "lightgray", title = "Firmicutes") + scale_x_continuous(limits = c(-25, 50)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Nitrospirae", fill = "lightgray", title = "Nitrospirae") + scale_x_continuous(limits = c(-1, 1)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Planctomycetes", fill = "lightgray", title = "Planctomycetes") + scale_x_continuous(limits = c(-10, 15)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Alphaproteobacteria", fill = "lightgray", title = "Alphaproteobacteria") + scale_x_continuous(limits = c(-25, 25)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Betaproteobacteria", fill = "lightgray", title = "Betaproteobacteria") + scale_x_continuous(limits = c(-10, 25)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Deltaproteobacteria", fill = "lightgray", title = "Deltaproteobacteria") + scale_x_continuous(limits = c(-10, 15)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Gammaproteobacteria", fill = "lightgray", title = "Gammaproteobacteria") + scale_x_continuous(limits = c(-25, 25)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Verrucomicrobia", fill = "lightgray", title = "Verrucomicrobia") + scale_x_continuous(limits = c(-15, 35)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Bacteria.Other", fill = "lightgray", title = "Bacteria.Other") + scale_x_continuous(limits = c(-10, 20)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "Archaea", fill = "lightgray", title = "Archaea") + scale_x_continuous(limits = c(-5, 5)) + stat_overlay_normal_density(color = "red", linetype = "dashed")
```

## Soil Chemistry

```{r}
#Soil Chemistry
shapiro.test(ttt.data.corr$pH)
shapiro.test(ttt.data.corr$EC)
shapiro.test(ttt.data.corr$GWC)
shapiro.test(ttt.data.corr$OC)

# Distribution of Soil Chemistry variables
ggdensity(ttt.data.corr, x = "pH", fill = "lightgray", title = "pH") + scale_x_continuous(limits = c(0, 10)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "EC", fill = "lightgray", title = "EC") + scale_x_continuous(limits = c(-100, 200)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "GWC", fill = "lightgray", title = "GWC") + scale_x_continuous(limits = c(-100, 200)) + stat_overlay_normal_density(color = "red", linetype = "dashed")

ggdensity(ttt.data.corr, x = "OC", fill = "lightgray", title = "OC") + scale_x_continuous(limits = c(-100, 100)) + stat_overlay_normal_density(color = "red", linetype = "dashed")
```

# Reproducibility

The session information is provided for full reproducibility.
```{r}
devtools::session_info()
```
