---
title: "Microbial Response to Intermittent Permafrost Thaw -- Soil Chemistry"
author: 'Authors: [Karl J. Romanowicz](https://kromanowicz.github.io/) and George W. Kling'
output:
  html_notebook:
    theme: spacelab
    toc: yes
    toc_depth: 5
    toc_float:
      collapsed: yes
      smooth_scroll: yes
  html_document:
    toc: yes
    toc_depth: '5'
    df_print: paged
  pdf_document:
    toc: yes
    toc_depth: '5'
---

__________________________________________________

**R Notebook:** <font color="green">Provides reproducible analysis for **Soil Chemistry** data in the following manuscript:</font>

**Citation:** Romanowicz KJ and Kling GW. (***In Press***) Summer thaw duration is a strong predictor of the soil microbiome and its response to permafrost thaw in arctic tundra. ***Environmental Microbiology***. [https://doi.org/10.1111/1462-2920.16218](https://doi.org/10.1111/1462-2920.16218)

**GitHub Repository:** [https://github.com/kromanowicz/2022-Annual-Thaw-Microbes](https://github.com/kromanowicz/2022-Annual-Thaw-Microbes)

**NCBI BioProject:** [https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA794857](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA794857)

**Accepted for Publication:** <font color="green">22 September 2022</font> *Environmental Microbiology*

# Experiment

This R Notebook provides complete reproducibility of the data analysis presented in ***"Summer thaw duration is a strong predictor of the soil microbiome and its response to permafrost thaw in arctic tundra"*** by Romanowicz and Kling.

<font color="green">This pipeline processes soil chemistry data related to the soil samples from which 16S rRNA gene sequences were generated using the Illumina MiSeq platform.</font>

```{r setup, include=FALSE}
# Set global options for notebook
knitr::opts_knit$set(root.dir = normalizePath("~/Desktop/TundraPro18"))
knitr::opts_chunk$set(echo = TRUE)
```

```{r message=FALSE, results='hide', warning=FALSE}
# Make a vector of required packages
required.packages <- c("corrr","data.table","devtools","dplyr","forcats","ggalluvial","ggdendro","ggplot2","ggpubr","grid","gridExtra","knitr","magrittr","microeco","patchwork","pheatmap","pvclust","qiime2R","RColorBrewer","tidyr","UpSetR","vegan")

# Load required packages
lapply(required.packages, library, character.only = TRUE)
```

```{r include=FALSE}
# set.seed is used to fix the random number generation to make the results repeatable
set.seed(123)
```

# Soil Chemistry

Import the metadata metrics
```{r}
# Load in the environmental metadata (qualitative and quantitative data)
asv.env <- read.csv("QIIME/R_Data/metadata.csv")

# Convert first column to row names
rownames(asv.env)<-asv.env[,1]
asv.env<-asv.env[,-1]

# Make dataframe
asv.env <- as.data.frame(asv.env)
```

## pH Plotting
```{r}
chem.plot<-asv.env

chem.plot$Increment<-factor(chem.plot$Increment, levels = c("90-100","80-90","70-80","60-70","50-60","40-50","30-40","20-30","10-20","0-10"))

pH.plot<-ggplot(data=chem.plot, aes(x=Increment, y=pH, group=site_tundra)) + geom_line(aes(linetype=site_tundra)) + geom_point(size=5, aes(shape=Tundra, fill=Site)) + scale_shape_manual("Tundra", values=c(21,24)) + coord_flip() + xlab("Soil Depth (cm)") + ylab("Soil pH") + theme_minimal() + theme(axis.line = element_line(colour = 'black', size = 1), axis.ticks = element_line(colour = "black", size = 2), axis.text.x = element_text(size = 14), axis.text.y = element_text(size = 14), panel.background = element_blank(), axis.title.x = element_text(size = 16), axis.title.y = element_text(size = 16), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.position = "none") + scale_y_continuous(expand = c(0, 0), limits = c(4.0, 7.1), position = "right") + guides(linetype = FALSE)

pH.plot

# Save as .png width=400, height=650 ("chem.pH.png")
```

## EC Plotting
```{r}
chem.plot$Increment<-factor(chem.plot$Increment, levels = c("90-100","80-90","70-80","60-70","50-60","40-50","30-40","20-30","10-20","0-10"))

EC.plot<-ggplot(data=chem.plot, aes(x=Increment, y=EC, group=site_tundra)) + geom_line(aes(linetype=site_tundra)) + geom_point(size=5, aes(shape=Tundra, fill=Site)) + scale_shape_manual("Tundra", values=c(21,24)) + coord_flip() + xlab("Soil Depth (cm)") + ylab("EC (uS/cm)") + theme_minimal() + theme(axis.line = element_line(colour = 'black', size = 1), axis.ticks = element_line(colour = "black", size = 2), axis.text.x = element_text(size = 14), axis.text.y = element_blank(), panel.background = element_blank(), axis.title.x = element_text(size = 16), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.position = "none") + scale_y_continuous(expand = c(0, 0), limits = c(0, 205), position = "right")

EC.plot

# Save as .png width=350, height=650 ("chem.EC.png")
```

## GWC Plotting
```{r}
chem.plot$Increment<-factor(chem.plot$Increment, levels = c("90-100","80-90","70-80","60-70","50-60","40-50","30-40","20-30","10-20","0-10"))

GWC.plot<-ggplot(data=chem.plot, aes(x=Increment, y=GWC, group=site_tundra)) + geom_line(aes(linetype=site_tundra)) + geom_point(size=5, aes(shape=Tundra, fill=Site)) + scale_shape_manual("Tundra", values=c(21,24)) + coord_flip() + xlab("Soil Depth (cm)") + ylab("GWC (%)") + theme_minimal() + theme(axis.line = element_line(colour = 'black', size = 1), axis.ticks = element_line(colour = "black", size = 2), axis.text.x = element_text(size = 14), axis.text.y = element_blank(), panel.background = element_blank(), axis.title.x = element_text(size = 16), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.position = "none") + scale_y_continuous(expand = c(0, 0), limits = c(0, 105), position = "right")

GWC.plot

# Save as .png width=350, height=650 ("chem.GWC.png")
```

## OC Plotting
```{r}
chem.plot$Increment<-factor(chem.plot$Increment, levels = c("90-100","80-90","70-80","60-70","50-60","40-50","30-40","20-30","10-20","0-10"))

OC.plot<-ggplot(data=chem.plot, aes(x=Increment, y=OC, group=site_tundra)) + geom_line(aes(linetype=site_tundra)) + geom_point(size=5, aes(shape=Tundra, fill=Site)) + scale_shape_manual("Tundra", values=c(21,24)) + coord_flip() + xlab("Soil Depth (cm)") + ylab("OC (%)") + theme_minimal() + theme(axis.line = element_line(colour = 'black', size = 1), axis.ticks = element_line(colour = "black", size = 2), axis.text.x = element_text(size = 14), axis.text.y = element_blank(), panel.background = element_blank(), axis.title.x = element_text(size = 16), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.text = element_text(size=14), legend.title = element_text(size=16)) + scale_y_continuous(expand = c(0, 0), limits = c(0, 52), position = "right") + guides(linetype = FALSE)

OC.plot

# Save as .png width=450, height=650 ("chem.OC.png")
```

Save the soil chemistry plots together
```{r}
soil.chem.plot <- (pH.plot | EC.plot | GWC.plot | OC.plot) + plot_annotation(tag_levels = "A")
soil.chem.plot

# Save as .eps image (width = 1000, height = 500; "soil.chem.eps")
```

# Statistics

## Metadata Stats

Statistics on Environmental Differences Across Sites
```{r}
#Create matrix of numeric columns
env.bind.mat<-as.matrix(cbind(asv.env[,8:12]))

# MANOVA
env.manova<-manova(env.bind.mat~site_tundra, data=asv.env)
env.man.sum<-summary.aov(env.manova)
env.man.sum
```

Run individual ANOVA on  environmental factors with significant site differences to get at Post-hoc analysis
```{r}
# pH ANOVA with Post-Hoc for Site Differences
env.pH<-aov(pH~Site*Tundra, data=asv.env)
TukeyHSD(env.pH)
```

```{r}
# EC ANOVA with Post-Hoc for Site Differences
env.EC<-aov(EC~Site*Tundra, data=asv.env)
TukeyHSD(env.EC)
```

```{r}
# GWC ANOVA with Post-Hoc for Site Differences
env.GWC<-aov(GWC~Site*Tundra, data=asv.env)
TukeyHSD(env.GWC)
```

```{r}
# OM ANOVA with Post-Hoc for Site Differences
env.OM<-aov(OM~Site*Tundra, data=asv.env)
TukeyHSD(env.OM)
```

```{r}
# OC ANOVA with Post-Hoc for Site Differences
env.OC<-aov(OC~Site*Tundra, data=asv.env)
TukeyHSD(env.OC)
```

Statistics on Environmental Differences by Soil Type (Org vs. Min)
```{r}
# MANOVA by Soil Type
env.manova.2<-manova(env.bind.mat~Soil, data=asv.env)
env.man.sum.2<-summary.aov(env.manova.2)
env.man.sum.2
```

# Reproducibility

The session information is provided for full reproducibility.
```{r}
devtools::session_info()
```